RFC/src/RFC-0201_TariScript.md

```diff
@@ -269,164 +264,162 @@ pub struct TransactionInput {
 ```
 
 In standard Mimblewimble, an input is the same as an output _sans_ range proof. The range proof doesn't need to be checked
-again when spending inputs, so it is dropped. However, under this scheme, the range proof must be checked again to ensure
-that the output metadata (script, output features) have not been modified.
+again when spending inputs, so it is dropped. 
+
 
-_Note:_ Currently, the output features are actually malleable. TariScript fixes this.
 
 ```rust,ignore
 pub struct TransactionInput {
     /// Options for an output's structure or use
     features: OutputFeatures,
     /// The homomorphic commitment representing the output amount
     commitment: Commitment,
-    /// A proof that the commitment is in the right range
-    proof: RangeProof,
     /// The serialised script
     script: Vec<u8>,
-    /// The public script key, L
-    script_pub_key: PublicKey
     /// The script input data, if any
     input_data: Vec<u8>
-    /// A signature with k', signing the script and input data
-    script_signature: Signature
+    /// A signature with K_b, signing the script and input data
+    script_signature: Signature,
+    /// The sender pubkey, K_a
+    sender_pub_key: PublicKey
+    /// The receiver pubkey, K_b
+    receiver_pub_key: PublicKey
 }
 ```
+The input data to the script is sign with the receiver pub key of the Input, as the person spending the input should have control of it. 
```

Comment:
This doesn't sound right. The sender owns the previous output, how can the receiver sign it. Is "the  person spending the input" not the Sender?

Reply:
The names can be very confusing. But on the input, the previous owner is the receiver. 

For example  Alice > BoB
C1 > C2

On C1, Alice is the reciever, and on C2, Alice is the sender. 
Even thou  in this TX,  Alice is the spender, spending it by sending it to Bob.

Reply:
I  see these values are actually the values in the previous output. It would probably be better for clarity to just reference the output via a hash